programs/stake/benches/stake_instructions.rs.bak

```diff
@@ -0,0 +1,159 @@
+        collections::HashSet,
+        str::FromStr,
+        sync::{Arc, RwLock},
+    },
+    test::Bencher,
+    test_case::test_case,
+};
+
+#[allow(dead_code)]
+struct TestSetup {
+    genesis_config: GenesisConfig,
+    mint_keypair: Keypair,
+    bank: Bank,
+    bank_forks: Arc<RwLock<BankForks>>,
+    amount: u64,
+
+    rent_address: Pubkey,
+    rent_account: AccountSharedData,
+    stake_account: AccountSharedData,
+}
+
+impl TestSetup {
+    fn new() -> Self {
+        let (mut genesis_config, mint_keypair) = create_genesis_config(sol_to_lamports(1.));
+        genesis_config.rent = Rent::default();
+        let (bank, bank_forks) = Bank::new_with_bank_forks_for_tests(&genesis_config);
+        let bank = Bank::new_from_parent(
+            bank,
+            &Pubkey::new_unique(),
+            genesis_config.epoch_schedule.get_first_slot_in_epoch(1),
+        );
+        let amount = genesis_config.rent.minimum_balance(0);
+
+        let rent = Rent::default();
+        let rent_exempt_reserve = rent.minimum_balance(StakeStateV2::size_of());
+        let rent_address = rent::id();
+        let rent_account = create_account_shared_data_for_test(&rent);
+        bank.store_account(&rent_address, &rent_account);
+
+        let stake_account = AccountSharedData::new(
+            rent_exempt_reserve,
+            StakeStateV2::size_of(),
+            &solana_stake_program::id(),
+        );
+
+        Self {
+            genesis_config,
+            mint_keypair,
+            bank,
+            bank_forks,
+            amount,
+
+            rent_address,
+            rent_account,
+            stake_account,
+        }
+    }
+
+    fn execute_init(&mut self) {
+        let stake_address = Pubkey::new_unique();
+        self.bank.store_account(&stake_address, &self.stake_account);
+
+        let instruction = solana_sdk::stake::instruction::initialize(
+            &stake_address,
+            &Authorized::auto(&stake_address),
+            &Lockup::default(),
+        );
+
+        let tx = Transaction::new(
+            &[&self.mint_keypair],
+            Message::new(&[instruction], Some(&self.mint_keypair.pubkey())),
+            self.genesis_config.hash(),
+        );
+
+        let batch = self.bank.prepare_batch_for_tests(vec![tx]);
+
+        let commit_result = self
+            .bank
+            .load_execute_and_commit_transactions(
+                &batch,
+                solana_sdk::clock::MAX_PROCESSING_AGE,
+                false,
+                ExecutionRecordingConfig::new_single_setting(false),
+                &mut ExecuteTimings::default(),
+                None,
+            )
+            .0
+            .remove(0);
+
+        assert!(commit_result.is_ok());
+        assert_eq!(commit_result.unwrap().status, Ok(()));
+    }
+}
+
+#[bench]
+fn bench_stake_transaction_initialize(bencher: &mut Bencher) {
+    let mut test_setup = TestSetup::new();
+
+    bencher.iter(|| {
+        test_setup.execute_init();
```

Comment:
Early attempt to push through test tx for full execution. It is too slow, too much overhead. I abandoned the approach, favor `mock_process_instruction`.